src/NHibernate.Test/TypesTest/DecimalTypeFixture.cs

```diff
@@ -10,9 +11,50 @@ namespace NHibernate.Test.TypesTest
 	[TestFixture]
 	public class DecimalTypeFixture : TypeFixtureBase
 	{
-		protected override string TypeName
+		protected override string TypeName => "Decimal";
+
+		private readonly DecimalType _type = NHibernateUtil.Decimal;
+		private const int _highScaleId = 2;
+		private const decimal _highScaleTestedValue = 123456789.123456789m;
+
+		protected override bool AppliesTo(Dialect.Dialect dialect)
+		{
+			return base.AppliesTo(dialect) && !TestDialect.HasBrokenDecimalType;
+		}
+
+		protected override void Configure(Configuration configuration)
+		{
+			base.Configure(configuration);
+
+			if (Dialect is FirebirdDialect)
+			{
+				configuration.SetProperty(Environment.ParameterDefaultDecimalPrecision, "18");
+				configuration.SetProperty(Environment.ParameterDefaultDecimalScale, "9");
```

Comment:
Due to Firebird needing an explicit cast in some queries, injected by its driver, we must set its default scale/precision according to what the application needs. (Applications having varying needs will be stuck by this Firebird limitation, unless they use separated session factories just for those varying needs...)